lib/aggregators/Aggregator.ts

```diff
@@ -0,0 +1,119 @@
+import type * as RDF from '@rdfjs/types';
+import * as RdfString from 'rdf-string';
+import type { Algebra } from 'sparqlalgebrajs';
+import type { ICompleteSharedContext } from '../evaluators/evaluatorHelpers/BaseExpressionEvaluator';
+import type * as E from '../expressions';
+import { TermTransformer } from '../transformers/TermTransformer';
+import type { ITermTransformer } from '../transformers/TermTransformer';
+import { TypeAlias } from '../util/Consts';
+import { isSubTypeOf } from '../util/TypeHandling';
+import type { IAggregatorComponentClass } from '.';
+
+export abstract class AggregatorComponent {
+  public abstract put(bindings: RDF.Term | undefined): void;
+  public abstract result(): RDF.Term | undefined;
+
+  protected separator: string;
+  protected termTransformer: ITermTransformer;
+
+  public static emptyValue(): RDF.Term | undefined {
+    return undefined;
+  }
+
+  public constructor(expr: Algebra.AggregateExpression, protected sharedContext: ICompleteSharedContext) {
+    this.separator = expr.separator || ' ';
+    this.termTransformer = new TermTransformer(sharedContext.superTypeProvider);
+  }
+
+  protected termToNumericOrError(term: RDF.Term): E.NumericLiteral {
+    // TODO: Check behaviour
```

Comment:
TODO

(consider adding a check in your editor that warn if there are TODO's before committing)

Reply:
This TODO was already there. I just left it here. I can remove it since it is tested? (100% coverage?)

Reply:
Ah ok. Let's remove then.